tests/backend/test_validators.py

```diff
@@ -0,0 +1,296 @@
+                "All obligatory metadata is not filled in for variables [{'pers_id': ['name']},"
+                in str(
+                    record[1].message,
+                )
+            )
+
+
+def test_obligatory_metadata_dataset_warning_name(metadata: Datadoc):
+    state.metadata = metadata
+    with pytest.warns(
+        ObligatoryDatasetWarning,
+        match="All obligatory metadata is not filled in for dataset",
+    ) as record:
+        metadata.write_metadata_document()
+    assert metadata.dataset.name is None
+    assert "name" in str(
+        record[0].message,
+    )
+
+    metadata.dataset.name = model.LanguageStringType(
+        [
+            model.LanguageStringTypeItem(languageCode="nb", languageText="Navnet"),
+        ],
+    )
+    with pytest.warns(
+        ObligatoryDatasetWarning,
+        match="All obligatory metadata is not filled in for dataset",
+    ) as record2:
+        metadata.write_metadata_document()
+    assert metadata.dataset.name is not None
+    assert "name" not in str(record2[0].message)
+
+    metadata.dataset.name = model.LanguageStringType(
+        [
+            model.LanguageStringTypeItem(languageCode="nb", languageText=""),
+        ],
+    )
+    with pytest.warns(
+        ObligatoryDatasetWarning,
+        match="All obligatory metadata is not filled in for dataset",
+    ) as record3:
+        metadata.write_metadata_document()
+    assert "name" in str(record3[0].message)
+
+
+def test_obligatory_metadata_dataset_warning_description(metadata: Datadoc):
+    state.metadata = metadata
+    missing_obligatory_dataset: str
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+        metadata.write_metadata_document()
+        if issubclass(w[0].category, ObligatoryDatasetWarning):
+            missing_obligatory_dataset = str(w[0].message)
+    assert metadata.dataset.description is None
+    assert re.search(r"\bdescription\b", missing_obligatory_dataset)
+    metadata.dataset.description = model.LanguageStringType(
+        [
+            model.LanguageStringTypeItem(languageCode="nb", languageText="Beskrivelse"),
+        ],
+    )
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+        metadata.write_metadata_document()
+        if issubclass(w[0].category, ObligatoryDatasetWarning):
+            missing_obligatory_dataset = str(w[0].message)
+    assert metadata.dataset.description is not None
+    assert not re.search(r"\bdescription\b", missing_obligatory_dataset)
+
+
+def test_obligatory_metadata_dataset_warning_description_multiple_languages(
+    metadata: Datadoc,
+):
+    state.metadata = metadata
+    missing_obligatory_dataset = ""
+
+    metadata.dataset.description = model.LanguageStringType(
+        [
+            model.LanguageStringTypeItem(languageCode="nb", languageText="Beskrivelse"),
+        ],
+    )
+    metadata.dataset.description = model.LanguageStringType(
```

Comment:
I can not say this is very useful as test, but it was written because of the issue with field names not being re-added to the list after the value was removed. I could probably refactor the tests even more now that the issue is solved